api/explorer.go

```diff
@@ -1,62 +1,334 @@
 package api
 
 import (
-	"fmt"
 	"net/http"
 
+	"github.com/NebulousLabs/Sia/build"
 	"github.com/NebulousLabs/Sia/types"
 )
 
-// Handles the call to get information about the blockchain. Returns
-// several data points such as chain height, the current block, and
-// file contract info
-func (srv *Server) explorerStatusHandler(w http.ResponseWriter, req *http.Request) {
-	writeJSON(w, srv.exp.ExplorerStatus())
+type (
+	// ExplorerBlock is a block with some extra information such as the id and
+	// height. This information is provided for programs that may not be
+	// complex enough to compute the ID on their own.
+	ExplorerBlock struct {
+		ID       types.BlockID     `json:"id"`
+		Height   types.BlockHeight `json:"height"`
+		RawBlock types.Block       `json:"rawblock"`
+	}
+
+	// ExplorerTransaction is a transcation with some extra information such as
+	// the parent block. This information is provided for programs that may not
+	// be complex enough to compute the extra information on their own.
+	ExplorerTransaction struct {
+		ID             types.TransactionID `json:"id"`
+		Parent         ExplorerBlock       `json:"parent"`
+		RawTransaction types.Transaction   `json:"rawtransaction"`
+	}
+
+	// ExplorerGET is the object returned as a response to a GET request to
+	// /explorer.
+	ExplorerGET struct {
+		// General consensus information.
+		Height            types.BlockHeight `json:"height"`
+		CurrentBlock      types.BlockID     `json:"currentblock"`
+		Target            types.Target      `json:"target"`
+		Difficulty        types.Currency    `json:"difficulty"`
+		MaturityTimestamp types.Timestamp   `json:"maturitytimestamp"`
+		Circulation       types.Currency    `json:"circulation"`
```

Comment:
hmm. `TotalCoins` maybe? A little closer to what the number actually represents.